### PR TITLE
detect: Propagate app-layer decode events

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #6435 
This PR propagates app-layer decode events so anomaly logging is possible.

Without this PR, file compression events and exceeding the inspection buffer limit are recorded but never exposed.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [4482](https://redmine.openinfosecfoundation.org/issues/4482)

Describe changes:
- Move un-exposed events to app-layer-events 
- Propagate app-layer decode events to packet
- Eliminates detection context event as events are now added to app layer event element of packet.

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.memuse | 537797248 | 509179328 | 105.62% |
| | generic_stats_chk |
| .flow.spare | 1986638 | 1866612 | 106.43% | 

Pipeline 5011

File: src/detect.c [hunk 1]
```diff
@@ -1288,6 +1288,9 @@ static void DetectRunTx(ThreadVars *tv,
     AppLayerGetTxIterState state;
     memset(&state, 0, sizeof(state));
 
+    /* Invariant across transactions */
+    det_ctx->p = p;
```
Comment (victorjulien) [inline src/detect.c:1292]: could move this to the general per packet detect entry point

Comment (jlucovsky) [reply src/detect.c:1292]: Like `DetectRunSetup` and `DetectRunCleanup`?

Comment (victorjulien) [reply src/detect.c:1292]: Yeah I think that would make sense.

**Comment by jlucovsky** [comment]
Continued in #6631

**Pull request closed without merging**